zircon-object/src/task/exception.rs

```diff
@@ -542,106 +506,73 @@ mod tests {
     #[test]
     fn exceptionate_iterator() {
         let parent_job = Job::root();
-        let job = parent_job.create_child(0).unwrap();
-        let proc = Process::create(&job, "proc", 0).unwrap();
-        let thread = Thread::create(&proc, "thread", 0).unwrap();
-
-        let exception = Exception::create(thread.clone(), ExceptionType::Synth, None);
-        let mut iterator = ExceptionateIterator::new(&exception);
-
-        assert!(Arc::ptr_eq(
-            &iterator.next().unwrap(),
-            &proc.get_debug_exceptionate()
-        ));
-        assert!(Arc::ptr_eq(
-            &iterator.next().unwrap(),
-            &thread.get_exceptionate()
-        ));
-        assert!(Arc::ptr_eq(
-            &iterator.next().unwrap(),
-            &proc.get_exceptionate()
-        ));
-        assert!(Arc::ptr_eq(
-            &iterator.next().unwrap(),
-            &job.get_exceptionate()
-        ));
-        assert!(Arc::ptr_eq(
-            &iterator.next().unwrap(),
-            &parent_job.get_exceptionate()
-        ));
-        assert!(iterator.next().is_none());
+        let job = parent_job.create_child().unwrap();
+        let proc = Process::create(&job, "proc").unwrap();
+        let thread = Thread::create(&proc, "thread").unwrap();
+
+        let exception = Exception::create(&thread, ExceptionType::Synth, None);
+        let iterator = ExceptionateIterator::new(&exception);
+        let expected = [
+            proc.debug_exceptionate(),
+            thread.exceptionate(),
+            proc.exceptionate(),
+            job.exceptionate(),
+            parent_job.exceptionate(),
+        ];
+        for (actual, expected) in iterator.zip(expected.iter()) {
+            assert!(Arc::ptr_eq(&actual, expected));
+        }
```

Comment:
This removes the test that the iterator does end after it generated expected items. (since the zipped iterator can ends before the tested iterator ends)
You can collect the iterator before zipping it and check the count of the collected items.